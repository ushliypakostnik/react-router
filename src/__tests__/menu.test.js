import React from 'react';

import Menu, { MenuItem } from '../components/menu';

describe('menu render', () => {
  const props = {
    data:[
      {
        key: 1,
        text: 'Item1',
        path: '/1',
        className: "class1",
        onClick: jest.fn()
      },
      {
        key: 2
      }
    ]
  };

  const wrapper = shallow(<Menu children={props.data} />);
  const item = shallow(<MenuItem {...props.data[0]} />);

  it('menu render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('menu has class', () => {
    expect(wrapper.hasClass('app__menu')).toEqual(true);
  });

  it('menu has two childs', () => {
    expect(wrapper.children()).toHaveLength(2);
  });

  it('item render correctly', () => {
    expect(toJson(item)).toMatchSnapshot();
  });

  it('item name is li', () => {
    expect(item.name()).toEqual('li');
  });

  it('item has class', () => {
    expect(item.find('Link').hasClass(props.data[0].className)).toEqual(true);
  });
});
