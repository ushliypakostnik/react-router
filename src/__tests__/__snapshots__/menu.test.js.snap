// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu render should render correctly 1`] = `
<ul
  className="app__menu test"
/>
`;
